src/forms/EditListingClassURLForm/EditListingClassURLForm.js

```diff
@@ -0,0 +1,153 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { arrayOf, bool, func, shape, string } from 'prop-types';
+import { compose } from 'redux';
+import { Form as FinalForm } from 'react-final-form';
+import { intlShape, injectIntl, FormattedMessage } from '../../util/reactIntl';
+import classNames from 'classnames';
+import { propTypes } from '../../util/types';
+import { required, composeValidators, validBusinessURL } from '../../util/validators';
+import { Form, Button, FieldTextInput } from '../../components';
+
+import css from './EditListingClassURLForm.module.css';
+
+const EditListingClassURLFormComponent = props => {
+  const { initialValues } = props;
+  const [numberOfURL, setNumberofURL] = useState(Object.keys(initialValues).length ? Object.keys(initialValues).length : 1);
```

Comment:
if the condition returns 0, null or undefined, we set the numberOfURL to 1 since there is no URL for this listing. Otherwise we set this to the number of URL currently available in this listing

Reply:
How does `Array.length` return null or undefined?

Reply:
My bad as I was only looking at the condition but not in this specific situation.
Array.length returns a number so if it's bigger than zero it will set the initial value to be that number.

As the null and undefined part, I was thinking of the situation when `initialValues` is not passed in or passed in wrong. I will patch this up by using Proptypes and checking before using the `Object.keys()` and `.length` on it.

Reply:
`Array.length returns a number so if it's bigger than zero it will set the initial value to be that number.`

So in this situation, we only have 2 possible outcomes that are the length is > 0 or not, what will you do to avoid the duplication of calling Object.keys twice?

Reply:
I just figured out a way it is to use this:
`const [numberOfURL, setNumberofURL] = useState(Object.keys(initialValues).length || 1);`
If the length is >0 it will return the number, otherwise it will return 1